src/github.com/couchbase/sync_gateway/rest/server_context.go

```diff
@@ -269,20 +269,49 @@ func (sc *ServerContext) getOrAddDatabaseFromConfig(config *DbConfig, useExistin
 
 	// Initialize event handlers, if any:
 	if config.EventHandlers != nil {
+
+		// Temporary solution to do validation of invalid event types in config.EventHandlers.
+		// config.EventHandlers is originally unmarshalled as interface{} so that we retain any
+		// invalid keys during the original config unmarshalling.  We validate the expected entries
+		// manually and throw an error for any invalid keys.  Then remarshal and
+		// unmarshal as EventHandlerConfig (considered manual reflection, but was too painful).  Comes with
+		// some overhead, but will only happen on startup/new config.
+		// Should be replaced when we implement full schema validation on config.
+
+		eventHandlers := &EventHandlerConfig{}
+		eventHandlersMap, ok := config.EventHandlers.(map[string]interface{})
+		if !ok {
+			return nil, errors.New(fmt.Sprintf("Unable to parse event_handlers definition in config for db %s", dbName))
+		}
+
+		// validate event-related keys
+		for k, _ := range eventHandlersMap {
+			if k != "max_processes" && k != "wait_for_process" && k != "document_changed" {
+				return nil, errors.New(fmt.Sprintf("Unsupported event property '%s' defined for db %s", k, dbName))
+			}
+		}
+
+		eventHandlersJSON, err := json.Marshal(eventHandlersMap)
+		err = json.Unmarshal(eventHandlersJSON, eventHandlers)
```

Comment:
it looks like this err is being ignored and will be overwritten by the resulting call to `json.Unmarshal`.  Even though it would probably create an error in the `json.Unmarshal` call, I think it will be easier to debug if the more specific error was logged.